src/fileformats/ocd_file_export.cpp

```diff
@@ -1369,28 +1436,24 @@ quint8 OcdFileExport::exportAreaSymbolCommon(const AreaSymbol* area_symbol, OcdA
 				pattern_symbol = pattern.point;
 				if (pattern.rotatable())
 					flags |= Ocd::SymbolRotatable;
+				first_point_pattern = &pattern;
 				break;
 			case Ocd::StructureAlignedRows:
-				ocd_area_common.structure_mode = Ocd::StructureShiftedRows;
-				// NOTE: This is only a heuristic which works for the
-				// orienteering symbol sets, not a general conversion.
-				// (Conversion is not generally possible.)
-				// No further checks are done to find out if the conversion
-				// is applicable because with these checks. Already a tiny
-				// (not noticeable) error in the symbol definition would make
-				// it take the wrong choice.
-				addWarning(::OpenOrienteering::OcdFileExport::tr("In area symbol \"%1\", assuming a \"shifted rows\" point pattern. This might be correct as well as incorrect.")
-				           .arg(area_symbol->getPlainTextName()));
-				
-				if (pattern.line_offset != 0)
+				if (!first_point_pattern)
+				{
+					qWarning("Unexpected sequence of area fill patterns");
+				}
+				else if (isStructureShiftedRows(*first_point_pattern, pattern))
+				{
+					ocd_area_common.structure_mode = Ocd::StructureShiftedRows;
 					ocd_area_common.structure_height /= 2;
-				else
-					ocd_area_common.structure_width /= 2;
-				
-				break;
+					break;
+				}
+				Q_FALLTHROUGH();
 			default:
 				addWarning(::OpenOrienteering::OcdFileExport::tr("In area symbol \"%1\", skipping a fill pattern.")
 				           .arg(area_symbol->getPlainTextName()));
+				/// \todo Export a single merged structure instead.
```

Comment:
Given that the TODO is removed in the next patch, I suggest removing it from the code and adding the mention about missing functionality into the commit message.

Reply:
Done.